package com.example.project.spring_caching_redis.Advices;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class APIResponse<T> {

//    @JsonFormat(pattern = "hh:mm:ss dd-MMM-YYYY")
    private LocalDateTime timeStamp;
    private T data;
    private APIError apiError;

    public APIResponse(){
        this.timeStamp = LocalDateTime.now();
    }

    public APIResponse(T data){
        this();
        this.data = data;
    }

    public APIResponse(APIError error){
        this();
        this.apiError = error;
    }
}

